src/backend/gl/src/tex.rs

```diff
@@ -220,10 +220,41 @@ pub fn format_to_glfull(format: NewFormat) -> Result<GLenum, ()> {
     })
 }
 
+fn determine_mipmap_level(desc: &t::Info, mip: Option<t::Mipmap>) -> t::Level {
+    use std::cmp::{min, max};
+    
+    match mip {
```

Comment:
please use the existing `get_num_levels` instead: https://github.com/gfx-rs/gfx/blob/7f30638ff6707bf473fb08e3ce3b2b87e92ebacd/src/core/src/texture.rs#L289